### DISCUSSION TITLE
Concourse handling destroy of old archived pipelines

### DISCUSSION THREAD

**Discussion by itsalix** [concourse/concourse#8903] [Ideas]
We manage tens of pipelines that set up to hundreds of other pipelines in Concourse, resulting in potentially thousands of archivable pipelines. When doing this via the set_pipeline method, the pipelines are archived indefinitely it seems, with pipelines that haven't been updated in years still existing, taking up a significant space in the database, even when the build logs have long passed the retention limit. (when manually cleared down we gained roughly 6MB of space per archived pipeline that was destroyed).
Could these pipelines be cleared down by Concourse itself, by updating the configuration to accept an environment variable such as CONCOURSE_MAX_DAYS_TO_RETAIN_ARCHIVED_PIPELINES. With the behaviour being that any archived pipelines last updated before the time limit set are automatically destroyed.
Would very much like to discuss this possibility, any input greatly appreciated 😃

**Comment by taylorsilva** [comment]
That's a good feature suggestion! I'm a bit surprised that Logging GC doesn't clear out logs, especially if you have a default logging GC set on your web nodes.

**Comment by itsalix** [comment]
Yeah we seem to find that when a pipeline is archived and the build logs pass the retention limits set, the archived pipeline still retains some information on the last successful run of a given job in that pipeline, and of course just other data about the pipeline can all add up when working with ludicrous numbers of pipelines
